k8s/templates/frontend-pgbouncer.yaml

```diff
@@ -0,0 +1,99 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "studio.fullname" . }}-dbpool-config
+  labels:
+    app: {{ template "studio.fullname" . }}
+data:
+  DB_HOST: {{ .Values.postgresql.externalCloudSQL.proxyHostName | default (include "postgresql.fullname" .) }}
+  # Aron: I configured pgbouncer based on the settings
+  # recommended by this article:
+  # https://medium.com/futuretech-industries/postgres-raffle-10k-connections-35-mo-part-two-b4c2e0c86e37
+  POOL_MODE: "transaction"
+  SERVER_RESET_QUERY: "DISCARD ALL"
+  # the custom listen backlog doesn't matter until we increase the nodes' somaxconn
+  LISTEN_BACKLOG: "16384"
+  DEFAULT_POOL_SIZE: !!str {{ .Values.studioApp.pgbouncer.pool_size | default 10 }}
+  RESERVE_POOL_SIZE: !!str {{ .Values.studioApp.pgbouncer.pool_size | default 5 }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "studio.fullname" . }}-dbpool-secrets
+  labels:
+    app: {{ template "studio.fullname" . }}
+data:
+  DB_USER: {{ .Values.postgresql.postgresUser | default "" | b64enc }}
+  DB_PASSWORD: {{ .Values.postgresql.postgresPassword | default "" | b64enc }}
+  DB_NAME: {{ .Values.postgresql.postgresDatabase | default "" | b64enc }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "studio.fullname" . }}-frontend-db-conn-pool-service
+  labels:
+    app: {{ template "studio.fullname" . }}
+spec:
+  ports:
+  - port: 5432
+    targetPort: 5432
+  selector:
+    app: {{ template "studio.fullname" . }}
+    tier: db
+    "pooler-for": frontend
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "studio.fullname" . }}-frontend-db-conn-pool
+  labels:
+    app: {{ template "studio.fullname" . }}
+    tier: db
+    "pooler-for": frontend
+spec:
+  replicas: {{ .Values.studioApp.pgbouncer.replicas | default 1 }}
```

Comment:
By default we set the replicas to 1. This is overriden in prod to at least 3 pods.